.github/workflows/cuda_exp.yml

```diff
@@ -0,0 +1,99 @@
+name: CUDA Experimental Version
```

Comment:
I think we don't need a new workflow which is mainly copy-paste from `cuda.yml`. Instead, it's possible to extend the current matrix and add new `task` env. variable:
https://github.com/microsoft/LightGBM/blob/af5b40e1f653e25a5f3470e09de4d1c68dc59663/.github/workflows/cuda.yml#L24-L37